pkg/apis/internal.acorn.io/v1/appinstance.go

```diff
@@ -159,10 +163,23 @@ type AppInstanceStatus struct {
 	AvailableAppImage      string                     `json:"availableAppImage,omitempty"`
 	ConfirmUpgradeAppImage string                     `json:"confirmUpgradeAppImage,omitempty"`
 	AppSpec                AppSpec                    `json:"appSpec,omitempty"`
+	Defaults               Defaults                   `json:"defaults,omitempty"`
+	Scheduling             map[string]Scheduling      `json:"scheduling,omitempty"`
 	Conditions             []Condition                `json:"conditions,omitempty"`
 	Endpoints              []Endpoint                 `json:"endpoints,omitempty"`
 }
 
+type Scheduling struct {
+	Requirements corev1.ResourceRequirements `json:"requirements,omitempty"`
+	Affinity     *corev1.Affinity            `json:"affinity,omitempty"`
+	Tolerations  []corev1.Toleration         `json:"tolerations,omitempty"`
+}
+
+type Defaults struct {
+	WorkloadClass *string           `json:"workloadClass,omitempty"`
```

Comment:
Do we need this field? After we calculate the defaults for a workload, why do we still need to know the workload class?

Related question: do we need the CPU on defaults as well? That way, the deployment won't change if the workload class changes.

Reply:
We shouldn't need defaults for the CPU as they are not recalculated unless the `Defaults.WorkloadClass` field changes. Given how the `defaults.Calculate` function works, this will not happen unless the app itself is updated. Similarly, `scheduling.Calculate` only runs on app updates as well and thus no recalculation of a change WorkloadClass will occur.